tests/test_authentication.py

```diff
@@ -140,13 +139,17 @@ def test_unknown_creds(self):
                               '^access denied: CredentialsLookupError: ')
 
     def test_alternative_credential_lookup(self):
+        # Temporarily use alternate auth
+        old_auth = self.auth
+        self.auth = AlternateAuthentication()
+
         sender = self._sender(credentials=ALTERNATIVE_CREDS)
         req = self._request(sender)
-        lookup_path = "%s.alternative_lookup" % __name__
-        with self.settings(HAWK_CREDENTIALS_LOOKUP=lookup_path):
```

Comment:
The other thing I changed, though, was to move the settings check outside `.authenticate()` (see [this change](https://github.com/kumar303/hawkrest/pull/34/files#diff-a8240a32be31704bcd6b7664afd0ec4bL51)) to avoid calling `import_string()` on each call to `authenticate()`. Seemed like needless overhead, and the only advantage it brought was being able to change the setting at runtime during testing.

With this change, though, the old test wouldn't work without `reload_module()` and all that.

So, three options:
1. I move the runtime check inside the default `_user_lookup()` method, with the `import_string()` overhead for every `authenticate()`. I don't like this option much, TBQH.
2. I leave the subclass-based test in place, and add back the `reload_module()`-based test I'd previously put in. It's ugly, agreed, but testing is where `reload_module()` is supposed to be used if at all.
3. Figure out some way to have a testing-only version of the call-time settings check (possibly by importing the default user/credential lookup functions explicitly, so those can be tested as well?), so the settings check can stay at import time normally and at runtime just for the test suite.